src/Behat/Gherkin/Filter/TagFilter.php

```diff
@@ -129,7 +126,7 @@ public function isScenarioMatch(FeatureNode $feature, ScenarioInterface $scenari
     /**
      * Checks that node matches condition.
      *
-     * @param string[] $tags
+     * @param list<string> $tags
```

Comment:
this change makes it more restrictive for callers (forcing them to provide a list) while the implementation does not care about keys. Why changing this type ?

Reply:
To provide good DX, it is better to be liberal in what you accept. My own rule is to use `list` in `@param` only when the method actually cares about lists (on the other hand, a `@return` that always returns a list should document it, to be more precise, which helps if the place using the return value cares about getting a list)

Reply:
I see your point, and I think that cases like this may be adjusted manually to be liberal (e.g. using `array<array-key, string>`).

The "reason" for this change is that it happened automatically with the following two rules:
- [phpdoc_list_type](https://cs.symfony.com/doc/rules/phpdoc/phpdoc_list_type.html)
- [phpdoc_array_type](https://cs.symfony.com/doc/rules/phpdoc/phpdoc_array_type.html)

I would keep those rules because I believe that we should avoid `T[]` or `array<T>` since they are severely misused. I believe those rules will nudge us in the right direction (case in point your comment).

What do you think? Is it too much effort to go this way?

Reply:
@stof some constructors, such as `src/Behat/Gherkin/Node/BackgroundNode.php`, also contain `list`-parameters. Should we also be liberal there?

Reply:
if the input is returned verbatim and we want to return a list, this is a legitimate use case for requiring a list in the parameter (as it means that the implementation cares about the keys in the input).
However, it should not be done automatically.

My proposal is to avoid enabling the risky rule `phpdoc_list_type` for now. Switching to list types is changing the actual signature of classes, and it is even more risky if we don't yet have static analysis checking the proper usages of lists.